app/assets/javascripts/workarea/storefront/paypal/modules/paypal.js

```diff
@@ -0,0 +1,54 @@
+/**
+ *
+ * Reliably ensure that the PayPal SDK is either loaded or will
+ * be loading.
+ *
+ * @namespace WORKAREA.paypal
+ */
+WORKAREA.registerModule('paypal', (function () {
+    'use strict';
+
+    var handlers = [],
+
+        /**
+         * Define a function that will be called when the paypal SDK has
+         * loaded.
+         *
+         * @method
+         * @name ready
+         * @param function handler - A function to be called when PayPal
+         *                           is available.
+         * @memberof WORKAREA.paypal
+         */
+        ready = function(handler) {
+            handlers.push(handler);
+        },
+
+        /**
+         * Continuously check for whether the Paypal SDK has loaded
+         * every 5 seconds, and call function handlers defined in
+         * `ready()` when the `window.paypal` object is available. This
+         * will only occur if the `<script>` tag containing the PayPal
+         * SDK is detected to be on the page.
+         *
+         * @method
+         * @name init
+         * @memberof WORKAREA.paypal
+         */
+        init = function() {
+            var $sdk = $('script[data-partner-attribution-id]'),
+                paypalLoaded = window.paypal !== undefined,
+                paypalExpectedToLoad = !_.isEmpty($sdk);
+
+            if (!paypalLoaded && paypalExpectedToLoad) {
+                window.setTimeout(init, 5000);
```

Comment:
is 5 seconds too long? Also, this should probably stop checking after so long. Like if the script becomes unavailable and never loads then this just keeps running. I think maybe 2 second timeout? With maybe a limit of doing that 5-10 times? Like if after 20(ish) seconds its not loaded its probably not going to.